beamer/tests/contracts/test_request_manager.py

```diff
@@ -676,6 +720,41 @@ def test_withdraw_without_challenge_with_resolution(
         request_manager.withdraw(claim_id, {"from": claimer})
 
 
+def test_withdraw_two_challengers(
+    request_manager, token, claim_stake, finalization_time, challenge_period_extension
+):
+    claimer, first_challenger, second_challenger, requester = alloc_accounts(4)
+    request_id = make_request(request_manager, token, requester, requester, 1)
+    claim = request_manager.claimRequest(request_id, 0, {"from": claimer, "value": claim_stake})
+    claim_id = claim.return_value
+
+    # First challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": first_challenger, "value": claim_stake + 1})
+    # Claimer outbids again
+    request_manager.challengeClaim(claim_id, {"from": claimer, "value": claim_stake + 1})
+    # Second challenger challenges
+    request_manager.challengeClaim(claim_id, {"from": second_challenger, "value": claim_stake + 1})
+
+    # Timetravel after claim period
+    chain.mine(timedelta=finalization_time + challenge_period_extension)
+
+    # Show that last challenger can withdraw first
+    request_manager.withdraw(claim_id, {"from": second_challenger})
+
+    # Challenger cannot withdraw twice
+    with brownie.reverts("Challenger has nothing to withdraw"):
+        request_manager.withdraw(claim_id, {"from": second_challenger})
+    with brownie.reverts("Challenger has nothing to withdraw"):
+        request_manager.withdraw(claim_id, {"from": claimer})
+
+    # First challenger withdraws his partial stake
+    request_manager.withdraw(claim_id, {"from": first_challenger})
+
+    # All stakes are withdrawn already
+    with brownie.reverts("Claim already withdrawn"):
+        request_manager.withdraw(claim_id, {"from": claimer})
+
+
 def test_withdraw_expired(token, request_manager):
     """Test that a request can be withdrawn once it is expired"""
     validity_period = 60 * 5
```

Comment:
We should also have a test with 2 or 3 challengers, with different stake amounts, to make sure that the amounts of ETH they withdraw at the end is correct. I haven't seen this covered.

Reply:
added the balance check to the test and changed stake amounts by the challengers to be different